dev/PipsControl/PipsControl.cpp

```diff
@@ -0,0 +1,369 @@
+    // TODO : Explain WHAT'S UP and mention quarks
+    if (!IsWithinBounds(args.GetCurrentPoint(*this).Position())) {
+        m_isPointerOver = false;
+        UpdateNavigationButtonVisualStates();
+        args.Handled(true);
+    }
+
+}
+
+bool PipsControl::IsWithinBounds(winrt::Point point) {
+    return point.X >= 0 && point.X <= ActualWidth() && point.Y >= 0 && point.Y <= ActualHeight();
+}
+
+void PipsControl::UpdateIndividualNavigationButtonVisualState(
+    bool hiddenOnEdgeCondition,
+    ButtonVisibility visibility,
+    const wstring_view visibleStateName,
+    const wstring_view hiddenStateName,
+    const wstring_view enabledStateName,
+    const wstring_view disabledStateName) {
+
+    if (visibility != ButtonVisibility::Collapsed) {
+        if ((visibility == ButtonVisibility::Visible || m_isPointerOver) && !hiddenOnEdgeCondition) {
+            winrt::VisualStateManager::GoToState(*this, visibleStateName, false);
+            winrt::VisualStateManager::GoToState(*this, enabledStateName, false);
+        }
+        else {
+            if (hiddenOnEdgeCondition) {
+                winrt::VisualStateManager::GoToState(*this, disabledStateName, false);
+            }
+            winrt::VisualStateManager::GoToState(*this, hiddenStateName, false);
+        }
+    }
+}
+
+void PipsControl::OnNavigationButtonVisibilityChanged(ButtonVisibility visibility, const wstring_view collapsedStateName) {
+    if (visibility == ButtonVisibility::Collapsed) {
+        winrt::VisualStateManager::GoToState(*this, collapsedStateName, false);
+    }
+    else {
+        UpdateNavigationButtonVisualStates();
+    }
+}
+
+void PipsControl::UpdateNavigationButtonVisualStates() {
+
+    const int selectedPageIndex = SelectedPageIndex();
+    const int numberOfPages = NumberOfPages();
+    const int maxDisplayedPages = MaxDisplayedPages();
+    
+    if (!numberOfPages == 0 && maxDisplayedPages > 0) { 
+        auto const ifPreviousButtonHiddenOnEdge = selectedPageIndex == 0;
+        UpdateIndividualNavigationButtonVisualState(ifPreviousButtonHiddenOnEdge, PreviousButtonVisibility(),
+                c_previousPageButtonVisibleVisualState, c_previousPageButtonHiddenVisualState,
+                c_previousPageButtonEnabledVisualState, c_previousPageButtonDisabledVisualState);
+
+        auto const ifNextButtonHiddenOnEdge = selectedPageIndex == numberOfPages - 1;
+        UpdateIndividualNavigationButtonVisualState(ifNextButtonHiddenOnEdge, NextButtonVisibility(),
+            c_nextPageButtonVisibleVisualState, c_nextPageButtonHiddenVisualState,
+            c_nextPageButtonEnabledVisualState, c_nextPageButtonDisabledVisualState);
+    }
+}
+
+void PipsControl::OnElementPrepared(winrt::ItemsRepeater sender, winrt::ItemsRepeaterElementPreparedEventArgs args)
+{
+    if (const auto pip = args.Element().try_as<winrt::Button>()) {
+        auto const pageNumber = unbox_value<int>(pip.Tag());
```

Comment:
>Tag [](start = 53, length = 3)

Why are we using the Tag property for this? ItemsRepeaterElementPreparedEventArgs has an index property